src/sysusers/sysusers.c

```diff
@@ -567,6 +571,18 @@ static int write_temporary_shadow(const char *shadow_path, FILE **tmpfile, char
                         return errno ? -errno : -EIO;
         }
 
+        /* Append the remaining NIS entries if any */
+        errno = 0;
```

Comment:
should be moved immediately before the putspent(), i.e. one line down, so that we know errno is zero-initialized every time we enter putspent)(

Reply:
Sorry but I don't see the point here. if putspent() fails, i.e. it returns -1, then errno should be set so I don't see why we need to reset it before the call.

And for probably the same reason, I dont see in case of failure of putspent() we need to test if errno is set or not before returning the error since it should always be set...

Reply:
so you do have a point. The main reason we do this errno munging here for putspent() is because the NSS APIs are generally pretty bad with setting errno (as the NSS modules themselves are responsible for that, and they pretty often are crap). putspent() is kinda NSS related even though not really NSS, hence if figure you could say we cargo culted this here...

Reply:
Hmm... okay... I take your word for granted and add an additional "errno = 0" right before putspent().

Nevertheless I think we should still let the "errno = 0" at line 575 in place since if the previous call to putspent() (line 570) succeed but set errno then we'll report the error.